middleware.js

```diff
@@ -1,34 +1,100 @@
 const vueCompiler = require('@vue/component-compiler')
+const fs = require('fs')
+const stat = require('util').promisify(fs.stat)
+const root = process.cwd()
+const path = require('path')
+const parseUrl = require('parseurl')
 const { transformModuleImports } = require('./transformModuleImports')
-const { loadPkg } = require("./loadPkg")
+const { loadPkg } = require('./loadPkg')
 const { readSource } = require('./readSource')
 
-const vueMiddleware = root => {
+const defaultOptions = {
+  cache: true
+}
+
+const vueMiddleware = (options = defaultOptions) => {
+  let cache
+  let time = {}
+  if (options.cache) {
+    const LRU = require('lru-cache')
+
+    cache = new LRU({
+      max: 500,
+      length: function (n, key) { return n * 2 + key.length }
+    })
+  }
+
   const compiler = vueCompiler.createDefaultCompiler()
 
   function send(res, source, mime) {
     res.setHeader('Content-Type', mime)
     res.end(source)
   }
 
+  async function tryCache (key, upd = true) {
```

Comment:
Can we use a more descriptive name than `upd`?

Reply:
(this should apply for all occurrences)

Reply:
@yyx990803 `updateTime`?

Reply:
@yyx990803 done